impeller/compiler/impellerc_main.cc

```diff
@@ -74,17 +75,42 @@ static bool OutputArtifacts(Compiler& compiler,
       std::cerr << "Could not create reflector." << std::endl;
       return false;
     }
-    auto stage_data = reflector->GetRuntimeStageData();
+    auto stage_data = reflector->GetRuntimeStageShaderData();
     if (!stage_data) {
       std::cerr << "Runtime stage information was nil." << std::endl;
       return false;
     }
-    if (sksl_mapping) {
-      stage_data->SetSkSLData(std::move(sksl_mapping));
+    RuntimeStageData stages;
+    if (sksl_shader) {
+      stages.AddShader(RuntimeStageBackend::kSkSL, sksl_shader);
     }
-    auto stage_data_mapping = options.json_format
-                                  ? stage_data->CreateJsonMapping()
-                                  : stage_data->CreateMapping();
+    switch (switches.target_platform) {
+      case TargetPlatform::kUnknown:
+      case TargetPlatform::kMetalDesktop:
+      case TargetPlatform::kMetalIOS:
+      case TargetPlatform::kOpenGLES:
+      case TargetPlatform::kOpenGLDesktop:
+      case TargetPlatform::kVulkan:
+        std::cerr << "TargetPlatform "
+                  << TargetPlatformToString(switches.target_platform)
+                  << " not supported for IPLR.";
+        return false;
+      case TargetPlatform::kRuntimeStageMetal:
+        stages.AddShader(RuntimeStageBackend::kMetal, stage_data);
+        break;
+      case TargetPlatform::kRuntimeStageGLES:
+        stages.AddShader(RuntimeStageBackend::kOpenGLES, stage_data);
+        break;
+      case TargetPlatform::kRuntimeStageVulkan:
+        stages.AddShader(RuntimeStageBackend::kVulkan, stage_data);
+        break;
```

Comment:
Just an FYI: once you get to the point of invoking the shader compiler multiple times for `--iplr` and packing these separately, you'll probably need to branch off and employ a similar approach to what happens for `--shader-bundle` and:
* Avoid generating intermediates for, reflection headers, etc.
* Remove all the unnecessary flags when `iplr` is enabled in the impellerc GN template.
* Relax the restrictions in switches.cc for `iplr` mode.

Reply:
I think I need to fix at least some of that as part of this PR. Looking into the failure on flutter_tester when we --enable-impeller